### ISSUE TITLE
[Syphilis] FHIR Mapping for Syphilis AOE data

### ISSUE THREAD

**Issue by DanielSass** [CDCgov/prime-simplereport#7564]
## Why should we do this?
- In order to report Syphilis data to STLTs we need to map the disease specific AOE questions to FHIR

## User story
As a SimpleReport engineer, I want syphilis AOE data sent in the appropriate FHIR format, so the data can flow through ReportStream and be available to STLTs

## Action Requested
- Update FHIR converter logic to include syphilis results and AOE responses in FHIR bundle

## Acceptance Criteria

**Review**
-   [ ] Reviewed with design and/or design lead
-   [ ] Reviewed with content strategist for final approval
-   [ ] Reviewed with product and engineering

## Additional Context
Syphilis AOE questions are listed in #7436 and are currently subject to change (4/16/2024).  Once they are finalized/confirmed we will need to coordinate with ReportStream to verify the appropriate FHIR format.

--
From Nicole Gallaway, CDC SME for Syphilis:

> _There is no ‘verified list of symptoms’. I think the signs/symptom list you show looks generally good, except alopecia appears missing._ 

Suggested order:

```
If Syphilis result [Radio button] Positive

What is the gender of their sexual partners? (Select all that apply)
[checkbox] Female
[checkbox] Male
[checkbox] Transwoman
[checkbox] Transman
[checkbox] Nonbinary or gender non-conforming
[checkbox] Gender identity not listed here
[checkbox] Prefer not to answer

Has the patient been told they have syphilis before?
[Radio button] Yes: 1087151000119108
[Radio button] No: 14732006
[Radio button] Prefer not to answer: 261665006

Is the patient pregnant?
[Radio button] Yes
[Radio button] No
[Radio button] Prefer not to say

Is the patient currently experiencing or showing signs of symptoms?
[Radio button] Yes
[Radio button] No

Which symptoms are they experiencing?
- Genital sore/lesion (SNOMED: 724386005)
- Anal sore/lesion (SNOMED: 195469007)
- Sore(s) in mouth/lips (SNOMED: 26284000)
- Body Rash (SNOMED: 266128007)
- Palmar (hand)/plantar (foot) rash (SNOMED: 56940005)
- Flat white warts (SNOMED: 91554004)
- Hearing loss (SNOMED: 15188001)
- Blurred vision (SNOMED: 246636008)
- Alopecia (SNOMED: 56317004)

Date of symptom onset
[MM/DD/YYYY]

[Button] Submit results
```

**Comment by jayna-SkylightDigital** [comment]
SNOMED codes (below) must be verified before we can proceed.

> SNOMED's for the requested symptoms
> 
> "Genital sore/lesion": "724386005" 
> 
> "Anal sore/lesion": "195469007" - Possible other values: Anal skin tag (195469007); Anal tag (195469007); Fibrous polyp of anus (195469007)
> 
> "Sore(s) in mouth/lips": "26284000" - Possible other values: Ulcer of mouth (26284000); Oral ulcer (26284000); Mouth ulceration (26284000); Ulceration of oral mucosa (26284000); Mouth ulcer (26284000)
> 
> "Body Rash": "266128007"
> 
> "Palmar (hand)/plantar (foot) rash": "56940005"
> 
> "Flat white warts": "91554004"
> 
> "Hearing loss": "15188001"
> 
> "Blurred vision": "246636008"
> 
> 
> 
> SNOMEDs for our “Has the patient been told they have syphilis before?"
> 
> "Yes": "1087151000119108"
> 
> "No": "14732006" -  This is the SNOMED for "No history of" for a general condition since we couldn't find one specific to syphilis
> 
> "Prefer not to answer": "261665006" - This is general “unknown” code, taken from the existing value set we have

**Comment by jayna-SkylightDigital** [comment]
From Victor Chaparro:

> _all AOE questions should be mapped to an Observation resource the same way we're receiving the other two AOE fields. And they should be reference under ServiceRequest.supportinInfo._

Slack: [https://skylight-hq.slack.com/archives/C0411VC78DN/p1695157667661819?thread_ts=1695156941.409009&cid=C0411VC78DN](https://skylight-hq.slack.com/archives/C0411VC78DN/p1695157667661819?thread_ts=1695156941.409009&cid=C0411VC78DN)


From Nicole Gallaway, CDC SME for Syphilis:

> _There is no ‘verified list of symptoms’. I think the signs/symptom list you show looks generally good, except alopecia appears missing._ 

**Symptoms**
- Genital sore/lesion: 724386005
- Anal sore/lesion: 195469007
- Sore(s) in mouth/lips: 26284000
- Body Rash: 266128007
- Palmar (hand)/plantar (foot) rash: 56940005
- Flat white warts: 91554004
- Hearing loss: 15188001
- Blurred vision: 246636008
- **Alopecia: 56317004** [[url](https://bioportal.bioontology.org/ontologies/SNOMEDCT?p=classes&conceptid=56317004)]

Has the patient been told they have syphilis before?
- Yes: 1087151000119108
- No: 14732006
- Prefer not to answer: 261665006

**Comment by DanielSass** [comment]
Need to coordinate with report stream to figure out how to map new AOE questions

**Comment by DanielSass** [comment]
New plan: punt the new AOE questions that we don't necessarily know how to map into a different ticket (#TBD) and map the more well understood ones within the scope of this ticket.

**Comment by jayna-SkylightDigital** [comment]
Please use AOE questions as shown in the above comment with the timestamp Aug 26, 2024 12:56 PM EDT.

**Please note the new question asking if the patient has been told they have syphilis in the past. This question will be present for Syphilis only.*

Has the patient been told they have syphilis before?
[Radio button] Yes: 1087151000119108
[Radio button] No: 14732006
[Radio button] Prefer not to answer: 261665006

**And the revised/update text for symptoms asking if they are experiencing or showing signs of symptoms. This can be updated to display with the same text for all conditions.** 
Is the patient currently experiencing or showing signs of symptoms?
[Radio button] Yes
[Radio button] No

**Comment by DanielSass** [comment]
Map the new history of syphilis data in line with other AOE questions, reach out to report stream to verify it looks correct.

**Comment by jayna-SkylightDigital** [comment]
@DanielSass will reach out to James Gilmore to determine the Syphilis-specific SNOMED code for history of Syphilis.

**Comment by emyl3** [comment]
> @DanielSass will reach out to James Gilmore to determine the Syphilis-specific SNOMED code for history of Syphilis.

@DanielSass Wanted to see if there's any follow-up on this? The only convo I'm seeing in Slack is [this one](https://skylight-hq.slack.com/archives/C0411VC78DN/p1721861543092299?thread_ts=1721667117.540449&cid=C0411VC78DN). 🤔 But based on the convo, James didn't specify what the code for that question should be 👀

**Comment by emyl3** [comment]
un-assigning myself from this for now
